math/lcm.go

```diff
@@ -0,0 +1,49 @@
+// Formula from https://en.wikipedia.org/wiki/Least_common_multiple
+// Find the least common multiple of two integers
+// by using the greatest common divisor.
+
+package main
+
+import (
+	"fmt"
+)
+
+// abs: absolute value for an integer
+func abs(a int) int {
+	if a < 0 {
+		return -a
+	}
+	return a
+}
+
+// gcd: greatest common divisor -
+// the largest positive integer that divides both a and b
+func gcd(a, b int) int {
+	if b == 0 {
+		return a
+	}
+	return gcd(b, a%b)
+}
+
+// lcm: least common multiple -
+// the smallest positive integer that is divisible by both a and b
+func lcm(a, b int) int {
+	if a == 0 && b == 0 {
+		return 0
+	}
+	a, b = abs(a), abs(b)
+	return a * b / gcd(a, b)
+}
+
+func main() {
```

Comment:
This repository avoids the use of `main` functions now, and therefore we recommend creating test cases in a test file. For examples, please take a look at any other algorithms (just look at Gcd algorithms for example).